gym-service/src/test/java/org/example/integration/steps/auth/LoginStep.java

```diff
@@ -0,0 +1,96 @@
+package org.example.integration.steps.auth;
+
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import lombok.RequiredArgsConstructor;
+import org.example.config.CustomUserDetails;
+import org.example.entities.User;
+import org.example.integration.config.TestContext;
+import org.example.repositories.UserRepository;
+import org.json.JSONObject;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.transaction.annotation.Transactional;
+import org.example.services.JWTService;
+
+import java.io.UnsupportedEncodingException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@RequiredArgsConstructor
+@Transactional
+public class LoginStep {
+    private final JWTService jwtService;
+    private final UserRepository userRepository;
+    private final TestContext testContext;
+
+    private final MockMvc mockMvc;
+
+    private MvcResult mvcResult;
+
+    @Given("the trainer is registered with username: {string}")
+    public void theTrainerIsAuthenticated(String username) {
+        setToken(username);
+    }
+
+    @Given("the trainee is registered with username: {string}")
+    public void theTraineeIsAuthenticated(String username) {
+        setToken(username);
+    }
+
+    @When("I send a POST request to {string} with valid credentials: username: {string}, password: {string}")
+    public void UserLoginWithUsernameAndPassword(String url, String username, String password) throws Exception {
+        JSONObject request = new JSONObject();
+        request.put("username", username);
+        request.put("password", password);
+
+        this.mvcResult = mockMvc.perform(MockMvcRequestBuilders.post(url)
+                        .content(request.toString())
+                        .contentType("application/json"))
+                .andReturn();
+        System.out.println(mvcResult.getResponse().getContentAsString());
+    }
+
+    @When("I send a POST request to {string} with invalid credentials: username: {string}, password: {string}")
+    public void UserLoginWithInvalidCredentials(String url, String username, String password) throws Exception {
+        JSONObject request = new JSONObject();
```

Comment:
what is difference from "ValidCredentials"? Do not copy same code, if there's no difference use generic methods.
define "valid" or "invalid" on GIVEN instead